### ISSUE TITLE
Generated Ogg files are too short

### ISSUE THREAD

**Issue by dbieber** [jpemartins/speex.js#15]
The Ogg files generated by your demo in the Encoding some WAV to OGG section (http://www.jpemartins.com/speex.js/) are too short. Much of the original WAV audio is lost in the conversion to Ogg.

**Comment by jpemartins** [comment]
Hi,

Thanks a lot, for reporting this.

I seem to hear the whole clip, using the web page to encode/decode. Could you tell me you reproduced the issue i.e. file you used, sample rate and what you expect?

NOTE: The generated file is not fully compliant with Ogg spec, since I don't handle granule_pos. So if you are testing on a normal player, may be it is related to this issue.

**Comment by dbieber** [comment]
I'm playing the Ogg files back with VLC after encoding the samples (female.wav or wb_male_speex_21.wav) in Chrome.

**Comment by dbieber** [comment]
You're right though; if I use the webpage to decode back to wav, the file is OK.
(And I see now this issue with some explanation: https://github.com/jpemartins/speex.js/issues/13)

**Comment by jpemartins** [comment]
Yeap, the problem is granule_pos as I reported in #13 . In VLC the audio might get cut.

Though I haven't found the time to fix it yet.

**Comment by pgee70** [comment]
any progress on this issue??

**Comment by Anniywell** [comment]
Can you fix it recently?

**Comment by pgee70** [comment]
Hi
I ended up using another library - I modified one I downloaded, but was not able to find a URL for you.  It encodes on the fly into opus format, not speech - It seems to give better quality at lower bitrates - I can also record voice for an hour into 18MB - So that is also pretty cool.

Regards
Pete
